packages/diagrams/lib/src/show_date_picker.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'diagram_step.dart';
+
+const Duration _pauseDuration = Duration(seconds: 1);
+const Duration _openDuration = Duration(milliseconds: 300);
+final Duration _totalDuration = _pauseDuration + _pauseDuration + _openDuration + _pauseDuration;
+final GlobalKey _buttonKey = GlobalKey();
+
+// Note: In order for this diagram to be captured successfully, you need to
```

Comment:
Hi @Piinks --

This is a totally fair point and one I'd like some help in addressing. The general problem: `showDialog` uses the `showGeneralDialog` under the hood. `showGeneralDialog` uses `rootNavigator: true`, and that isn't configurable in any way.

`showGeneralDialog` doesn't do much itself, it simply pushes a new route to the rootNavigator stack. However, it pushes a Route that is private: `_DialogRoute`, so we can't just push that ourselves directly.

I can see three options:
  
  1. Create a PR against Flutter that allows developers to override this option. In practice, this means `shareDialog` and `shareGeneralDialog` would accept `rootNavigator` as an optional parameter, with the default set as `true`.
  2. Make `_DialogRoute` public (`DialogRoute`) and add appropriate documentation for it to be used independently. Then, instead of using `showDialog`, we could just push the `DialogRoute` ourselves from this codebase. I'm not sure this is the best way to go, since `showDialog` could add additional functionality in the future.
  3. Go with your suggestion, keep them in main but comment them out in a special section by themselves.

I'm not 100% sure what the best course of action is, but I think option number 1 might be a good option since it looks like other projects also desire this functionality and this change could be introduced in a non-breaking way: https://github.com/flutter/flutter/issues/23565

Reply:
Hey there! I'm going to loop in @goderbauer to see what he thinks of your ideas. I am about to be away for a few days and didn't want to leave you waiting. I do agree that option 3 is less than optimal. It is also helpful that we have an issue here that shows it may be a good change to make.

Reply:
We shouldn't check this in as-is for the reasons that @Piinks pointed out. Of the options proposed, 1) sounds the best so far. However, I am also wondering: Why is there a Navigator present at all in the digram generator setup? If that Navigator wouldn't exist, the one introduced here would be the root and everything would just work out.

Reply:
Hi @goderbauer -- thanks for the feedback. The reason this app has a root navigator: The Diagram Capture tool sets up a `MaterialApp`, which in turns sets up localizations, overlays, directionality widgets, and more that are required by many Widgets.

I've updated this PR to remove the `MaterialApp` and replaced it with the Widgets necessary to run the app. In some ways, this feels precarious because `MaterialApp` may introduce a new Widget in the future which is required to render the UI, but it would solve the problem without needing any changes to Flutter core.

What are your thoughts about the latest changes?

Reply:
Yeah, it is a little precarious. Maybe we should just change showGeneralDialog and friends to take an optional NavigatorState. If it is provided, that's the Navigator on which we will push the dialog. If not, you'll get the root Navigator. That seems to offer the most flexibility.

Reply:
Thanks again, @goderbauer. I've created a PR against flutter that adds a `navigator` option to `showDialog`, `showDatePicker`, and `showGenericDialog` and updated this PR to use that technique to demonstrate how it would solve the problem. 

When you have a moment, please have a look at the PR and the changes included here to see if it results in a satisfactory API for Flutter and solution for this particular problem.

https://github.com/flutter/flutter/pull/42842/files